wagtail_review/static_src/admin/src/api/reviewer.ts

```diff
@@ -0,0 +1,103 @@
+export interface UserApi {
+    id: number | string;
+    full_name: string;
+    username: string;
+}
+
+export interface ExternalReviewerApi {
+    email: string;
+}
+
+interface ReviewerApi_Internal {
+    id: number;
+    internal: UserApi;
+    external: null;
+}
+
+interface ReviewerApi_External {
+    id: number;
+    internal: null;
+    external: ExternalReviewerApi;
+}
+
+export type ReviewerApi = ReviewerApi_Internal | ReviewerApi_External;
+
+export interface NewReviewerSuccess {
+    status: 'ok';
+    reviewer: ReviewerApi;
+}
+
+export interface NewReviewerValidationError {
+    status: 'error';
+    email?: string;
+}
+
+export type NewReviewerResponse =
+    | NewReviewerSuccess
+    | NewReviewerValidationError;
+
+export default class ReviewerAPIClient {
+    async getUsers({ search }: { search?: string }): Promise<UserApi[]> {
+        let params = [];
+
+        if (search) {
+            params.push(`search=${encodeURIComponent(search)}`);
+        }
+
+        const paramsStr = params ? '?' + params.join('&') : '';
+
+        let response = await fetch(
+            `/admin/wagtail_review/api/users/${paramsStr}`,
+            {
+                credentials: 'same-origin'
+            }
+        );
+
+        return response.json();
+    }
+
+    async _newReviwer(body: string): Promise<NewReviewerResponse> {
+        let response = await fetch(`/admin/wagtail_review/api/reviewers/`, {
+            method: 'POST',
```

Comment:
The `AdminAPIViewMixin` relies on session authentication, which according to https://www.django-rest-framework.org/api-guide/authentication/#sessionauthentication will require a valid CSRF token for any POST requests as here. Is this attached somehow? Or does `FooAuth` circumvent this, in which case there's maybe a case for this being somewhat unsafe?

Reply:
The fact it's called `FooAuth` suggests to me this was only a temporary hack that I've forgotten about. Thanks for spotting!

Reply:
Fixed in latest commit